# [ DATA SCIENCE ] Assignment
## [ CHI-SQUARE TEST ]
## S.AKHIL
### Gmail : akhilsangamrjasn@gmail.com

## 1. State the Hypotheses

## H0: The type of smart home device (Smart Thermostat vs. Smart Light) is independent of customer satisfaction level.
 
## Ha: The type of smart home device and customer satisfaction level are associated


H0 (Null Hypothesis): Imagine we're saying, "Whether a customer likes a smart thermostat or a smart light, their satisfaction level doesn't change. The type of device doesn't make a difference to how happy they are."

Ha (Alternative Hypothesis): On the flip side, we're suggesting, "The kind of smart home device a customer uses (smart thermostat vs. smart light) does affect their happiness. The type of device is linked to how satisfied they feel."

## 2. Compute the Chi-Square Statistic:

 Here Satisfaction of Smart Thermostat is represnted by set A and Smart Light by set B

In [101]:
import numpy as np
import scipy.stats as stats

In [103]:
x=[50,80,60,30,20]
y=[70,100,90,50,50]
## to calculate the mean of the data
x_mean = np.mean(x)
y_mean= np.mean(y)
print(x_mean,y_mean)

48.0 72.0


In [105]:
# Observed frequencies
obs = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

In [107]:
# Row and column totals
row_totals = obs.sum(axis=1) # axis = 1 is for rows
col_totals = obs.sum(axis=0) # axis = 0 is for columns
grand_total = obs.sum()

In [109]:
# Expected frequencies
exp_f= np.outer(row_totals, col_totals) / grand_total

In [111]:
# Compute Chi-Square Statistic
chi_square_stat = ((obs - exp_f) ** 2 / exp_f).sum()

In [113]:
# Degrees of freedom
rows, cols = obs.shape
df = (rows - 1) * (cols - 1)

In [115]:
# Critical value at alpha = 0.05
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, df)

In [117]:
# p-value
p_value = stats.chi2.sf(chi_square_stat, df)

In [119]:
# Decision
if chi_square_stat > critical_value:
    conclusion = "Reject the null hypothesis (H0)."
else:
     conclusion = "Fail to reject the null hypothesis (H0)."

In [121]:
# Output results
print("Chi-Square Statistic:", round(chi_square_stat, 2))
print("Degrees of Freedom:", df)
print("Critical Value:", round(critical_value, 2))
print("p-value:", round(p_value, 4))
print("Decision:", conclusion)

Chi-Square Statistic: 5.64
Degrees of Freedom: 4
Critical Value: 9.49
p-value: 0.2278
Decision: Fail to reject the null hypothesis (H0).


## 3.The Critical Value Using the alpha of 0.05 and the degrees of freedom is 9.49

## DECISION : 

FAIL TO REJECT THE NULL HYPOTHESIS

## Steps to Perform the  Chi-Square Statistic value

STEP_1: Calculate the expected frequencies for each cell using the formula:
(Row_Total * Column_Total) / (Grand_Total)

STEP_2: computing the chi-Square Statistic

Step_3: Determine the critical Value:
        The degrees of Freedom(df) are calculates as: 
            df = (number_of_rows-1) * (number_of_columns-1)
            Alpha = 0.05

STEP_4: compare the computed X Square value to the critical value
            If X Square Value is Greater than Critical_Value then Reject H0
            otherwise
            Fail to Reject the H0

## Values 

1. Chi-Square Statistic: 5.64
2. Degrees of Freedom: 4
3. Critical Value: 9.49
4. p-value: 0.2278
5. Decision: Fail to reject the null hypothesis (H0).

## CONCLUSION

#### After performing the Chi-Square test for independence, the results are as follows:

Chi-Square Statistic: Measures how much the observed data differs from what we expected.

Critical Value: A specific value from the Chi-Square table based on a significance level (alpha = 0.05) and degrees of freedom.

Decision: We compare the Chi-Square statistic to the critical value to make our conclusion.

#### Based on the analysis:
We fail to reject the null hypothesis (𝐻0).

This means there is no significant association between the type of smart home device (Smart Thermostat vs. Smart Light) and customer satisfaction level.

Interpretation: Customer satisfaction levels are similar regardless  or do not significantly differ,whether they bought a Smart Thermostat or a Smart Light.